src/components/ModalPayment/SquareModal.tsx

```diff
@@ -61,117 +56,113 @@ const ModalPayment = ({
     return makeSquarePayment( nonce, payment, buyer )
       .then((res) => {
         if (res.status === 200) {
-          setIsShown(true)
-        } else {
-          setIsShown(false)
+          dispatch({type: SET_MODAL_VIEW, payload: 2})
+        } 
+      })
+      .catch((err) => {
+        if( err.response
+          && err.response.data
+          && err.response.data.errors
+          && err.response.data.errors.length > 0){          
+          const responseErrors = err.response.data.errors
+          const newErrors = errorMessages.length > 0 ? [...errorMessages,responseErrors.map((error: { detail: string }) => error.detail) ] : responseErrors.map((error: { detail: string }) => error.detail)
```

Comment:
In the future we probably want to set custom error messages either on the client or server vs using Square's. We can keep this in mind as a follow up